diff --git a/Core/Inc/hm01b0.h b/Core/Inc/hm01b0.h
index 9ce4e95..1cee4e8 100644
--- a/Core/Inc/hm01b0.h
+++ b/Core/Inc/hm01b0.h
@@ -8,6 +8,9 @@
 #ifndef INC_HM01B0_H_
 #define INC_HM01B0_H_
 
+void HM01B0_WriteRegister(I2C_HandleTypeDef *hi2c, uint16_t reg_addr, uint8_t value);
+uint8_t HM01B0_ReadRegister(I2C_HandleTypeDef *hi2c, uint16_t reg_addr);
+void HM01B0_Init(void);
 
 
 #endif /* INC_HM01B0_H_ */
diff --git a/Core/Src/hm01b0.c b/Core/Src/hm01b0.c
index d6ffeb3..09ae569 100644
--- a/Core/Src/hm01b0.c
+++ b/Core/Src/hm01b0.c
@@ -29,6 +29,10 @@ uint8_t HM01B0_ReadRegister(I2C_HandleTypeDef *hi2c, uint16_t reg_addr) {
     return value;
 }
 
+void HM01B0_Init(void) {
+    // Example: read a read only register
+    uint8_t reg_value = HM01B0_ReadRegister(&hi2c2, SILICON_REV);
+}
 
 
 /*
diff --git a/Core/Src/main.c b/Core/Src/main.c
index faa1a28..4ee531a 100644
--- a/Core/Src/main.c
+++ b/Core/Src/main.c
@@ -104,22 +104,9 @@ int main(void)
   MX_I2C2_Init();
   MX_TIM2_Init();
   /* USER CODE BEGIN 2 */
-// '''''''''''''''''''''''''' 30/10/2024 ''''''''''''''''''''''''''''''''''''
-  void HM01B0_Init(void) {
-      // Example: write to a specific register to initialise the camera
-      uint8_t data[2]; // Buffer for register address and data
-      data[0] = HM01B0_REGISTER_X;  // Replace with the actual register address
-      data[1] = HM01B0_INIT_VALUE;   // Replace with the value you want to write
-
-      if (HAL_I2C_Master_Transmit(&hi2c2, HIMAX_I2C_ADDR, data, 2, HAL_MAX_DELAY) != HAL_OK) {
-          // Handle error
-          Error_Handler();
-      }
-  }
-
-  // '''''''''''''''''''''''''' 30/10/2024 ''''''''''''''''''''''''''''''''''''
   HM01B0_Init();
-
+  HM01B0_WriteRegister(&hi2c2, IMG_ORIENTATION, 0xAB);
+  uint8_t reg_value = HM01B0_ReadRegister(&hi2c2, IMG_ORIENTATION);
 
   /* USER CODE END 2 */
 
